index.ts

```diff
@@ -30,3 +30,42 @@ export function testState<T>(
 
   return state;
 }
+
+let stateId = 0;
+function genStateId(): number {
+  stateId += 1;
+  return stateId;
+}
+
+const STATE_ID_PROPERTY = "___testStateStateId";
+
+let stateIdToState: Map<number, Record<string, unknown>> = new Map();
+
+function deepCopy<T>(v: T): T {
+  return JSON.parse(JSON.stringify(v)) as T;
+}
+
+export function resetState(state: Record<string, unknown>): void {
+  if (STATE_ID_PROPERTY in state) {
+    const stateId = (state as any)[STATE_ID_PROPERTY];
+    const initState = deepCopy(stateIdToState.get(stateId));
+    if (initState === undefined) {
+      throw new Error("Failed to get initial state.");
+    }
+
+    for (const key in state) {
+      state[key] = initState[key];
+    }
+
+    return;
+  }
```

Comment:
I think you can split the body of this function into 2 separate functions e.g.:
`resetToExistingState` and `initializeState`.
So it would look like this:
```typescript
// function resetState
if (STATE_ID_PROPERTY in state) {
  return resetToExistingState(state);
}
return initializeState(state);
```

Reply:
Applied, added one more method and slightly adjusted naming - please take a look